specs/merkle-tree.md

```diff
@@ -0,0 +1,142 @@
+
+So for example, if we intended to run a single transaction on Goerli (5), OP Goerli (420), and Polygon Mumbai (80001). Then the leaves are are expected to be ordered like so:
+1. Approval Leaf on Goerli (Chain ID 5)
+2. Execute Leaf on Goerli
+3. Approval Leaf on OP Goerli (Chain ID 420)
+4. Execute Leaf on OP Goerli
+5. Approval Leaf on Polyon Mumbai (Chain ID 80001)
+6. Execute Leaf on Polygon Mumbai
+
+## Merkle Tree Generation Logic
+We provide utility for generating Sphinx Merkle trees. This utility is used by the official Sphinx Foundry plugin. We expect it to generate Merkle trees that adhere to the above invariants.
+
+#### `const makeSphinxMerkleTree = (deploymentData: DeploymentData): SphinxMerkleTree`
+
+#### Input
+Accepts a [DeploymentData](https://github.com/sphinx-labs/sphinx/blob/591ff8a33578ecf61f184f4afc83a551397009f9/packages/contracts/src/merkle-tree.ts#L45C23-L45C23) object where the keys are canonical chain ids and the values are deployment data objects which contain all of the necessary info to assemble a SphinxMerkleTree. The Sphinx Merkle tree generation function should be agnostic to the source of the transaction data so it can be used to assemble a tree based on transactions generated from any scripting framework (i.e Foundry, Hardhat Ignition, some arbitrary future framework).
+
+#### Output
+Outputs a [SphinxMerkleTree](https://github.com/sphinx-labs/sphinx/blob/591ff8a33578ecf61f184f4afc83a551397009f9/packages/contracts/src/merkle-tree.ts#L115) object that follows the above specification and invariants.
+
+## Dependencies
+The Merkle tree generation logic makes calls to two battle-tested external libraries. We use `ethers` version 6.7.0 to handle ABI encoding tree leaf `data`, and we use `@openzeppelin/merkle-tree` version 1.0.5 to assemble the Merkle tree. We test that interactions with these libraries work properly in our [SphinxModuleProxy unit tests](https://github.com/sphinx-labs/sphinx/blob/feature/pre-audit/packages/contracts/test/SphinxModuleProxy.t.sol) which cover that Merkle trees generated by this logic have data that is encoded properly and that they are executable on-chain. However, we do not test the internals of these libraries and instead rely on the assumption that they are robust and bug-free.
+
+## Assumptions
+The Merkle tree generation logic outputs trees which are valid and executable. However, there is no foolproof requirement that it be used. We make the asumption that relayers like the Sphinx DevOps platform will refuse to execute Merkle trees that are not valid according to the invariants defined in this document.
```

Comment:
> The Merkle tree generation logic outputs trees which are valid and executable.

Saying they're "valid" is a bit redundant because, by definition, a valid Merkle tree adheres to this spec (according to the vocabulary section). Saying they're "executable" is a bit confusing because there may be on-chain conditions that prevent the Merkle tree from being executed on-chain, which is an assumption that we discuss shortly after this